psi4/driver/schema_wrapper.py

```diff
@@ -586,6 +586,18 @@ def run_json_qcschema(json_data, clean, json_serialization, keep_wfn=False):
             )):
                 json_data["extras"]["qcvars"][k] = _serial_translation(v, json=json_serialization)
 
+    # Add in handling of matrix arguments which need to be obtained by a
+    # a function call.
+
+    if "psi4:arrays" in json_data["extras"] :
+        if "tIjAb" in json_data["extras"]["psi4:arrays"]:
+            json_data["extras"]["psi4:arrays"]["tIjAb"] = wfn.get_amplitudes()["tIjAb"].to_array().tolist()
+        if "tIA" in json_data["extras"]["psi4:arrays"]:
+            json_data["extras"]["psi4:arrays"]["tIA"] = wfn.get_amplitudes()["tIA"].to_array().tolist()
+        if "Da" in json_data["extras"]["psi4:arrays"]:
```

Comment:
I'm nervous about this because densities aren't handled sanely across all Psi modules. I'm not happy about density handling even within the `cc` megamodule and am in the slow process of refactoring. Lori, is it possible/advisable to only have this file for specific modules where we've verified the densities?

Reply:
densities are indeed a mess, thanks a ton for chipping away at that problem. generally correlated densities (especially `cc` and non-DF `mp2`) don't get updated unless you ask for an `OEProp` or response property, and there are probably other nasty specifics elsewhere. however, seeing as how this is in `extras`, I would qualify this as an "expert" option, so I think leaving a clarifying comment here in the harness would be sufficient (unless there's another place we're documenting these entries).

Reply:
I think having a list a methods of verified sanity and then only saving if `json_data["model"]["method"]` in that list is cautious and good. Have this "opt-in" for a while.